tests/ui/from_iter_instead_of_collect.rs

```diff
@@ -10,4 +10,7 @@ fn main() {
     HashMap::<usize, &i8>::from_iter(vec![5, 5, 5, 5].iter().enumerate());
 
     Vec::from_iter(vec![42u32]);
+
+    let a = vec![0, 1, 2];
+    assert_eq!(a, Vec::from_iter(0..3));
```

Comment:
How about this case? The suggested code seems not to work.
```
    let mut b = VecDeque::from_iter(0..3);
    // let mut b = `(0..3).collect::<Vec<_>>()` // the suggested code
    b.push_back(4);
```

Reply:
Yes indeed, turbofish should be customized to match the type implementing `from_iter` (here `VecDeque`)

Reply:
Looks better now.
I wanted to shorten the suggestion from `(0..3).collect::<std::collections::VecDeque<i32>>()` to `(0..3).collect::<VecDeque<i32>>()` by removing unneeded `std::collections` but I am not sure this is even possible so I need to dig more into this.

Reply:
I can't find any util method. IMHO it might be good to just convert `ty` to `String` and use `.split("::").last().unwrap()` if `.contains("::")`.

Reply:
Yeah I thought about this also, but it sounds a bit too hacky :smile:
Ultimately I will consider it if necessary.

Reply:
Okay, and yes it's a little hacky. FYI similar codes already exist here https://github.com/rust-lang/rust-clippy/blob/e5a1f22f486d70676620cc67cad45ecf359fb56e/clippy_lints/src/macro_use.rs#L79-L88.

Reply:
Okay, and yes it's a little hacky. FYI similar codes already exist here https://github.com/rust-lang/rust-clippy/blob/e5a1f22f486d70676620cc67cad45ecf359fb56e/clippy_lints/src/macro_use.rs#L79-L88.

Reply:
I agree, using items in scope to shorten the paths seems like a worthwile improvement. However, I don't think that should block this PR.

So I'd r+ this unless you want to continue improving it, @ThibsG.

Reply:
I want to give a try to improve this. I should be able to work on this really soon, and I keep you posted :wink:

Reply:
Try `cx.tcx.trimmed_def_paths(LOCAL_CRATE).get(&def_id)`